ports/stm32/boards/NUCLEO_WB55/rfcore_makefirmware.py

```diff
@@ -41,27 +44,149 @@
     "stm32wb5x_FUS_fw.bin": "fus_110.bin",
     "stm32wb5x_BLE_HCILayer_fw.bin": "ws_ble_hci.bin",
 }
+_RELEASE_NOTES = "Release_Notes.html"
+
+
+def get_details(release_notes, filename):
+    if not release_notes:
+        return None
+    file_details = re.findall(
+        rb"%s,(((0x[\d\S]+?),)+[vV][\d\.]+)[<,]" % filename.encode(),
+        release_notes,
+        flags=re.DOTALL,
+    )
+    # The release note has all past version details also, but current is at top
+    latest_details = file_details[0][0].split(b",")
+    addr_1m, addr_640k, addr_512k, addr_256k, version = latest_details
+    addr_1m = int(addr_1m, 0)
+    addr_640k = int(addr_640k, 0)
+    addr_512k = int(addr_512k, 0)
+    addr_256k = int(addr_256k, 0)
+    version = [int(v) for v in version.lower().lstrip(b"v").split(b".")]
+    return addr_1m, addr_640k, addr_512k, addr_256k, version
 
 
 def main(src_path, dest_path):
-    for src_file, dest_file in _FIRMWARE_FILES.items():
-        src_file = os.path.join(src_path, src_file)
+
+    # Load the release note to parse for important details
+    with open(os.path.join(src_path, _RELEASE_NOTES), "rb") as f:
+        release_notes = f.read()
+    # remove some formatting
+    release_notes = re.sub(rb"</?strong>", b"", release_notes)
+    # flatten tables
+    release_notes = re.sub(
+        rb"</t[dh]>\W*\n*\W*",
+        b",",
+        release_notes.replace(b"<td>", b"").replace(b"<th>", b""),
+    )
+    if (
+        b"Wireless Coprocessor Binary,STM32WB5xxG(1M),STM32WB5xxY(640k),STM32WB5xxE(512K),STM32WB5xxC(256K),Version,"
+        not in release_notes
+    ):
+        print(
+            "Cannot determine binary load address, please confirm manually in Release Notes."
+        )
+        release_notes = None
```

Comment:
Should this terminate? It looks like `release_notes` being None will just crash when the return value of `get_details` is unpacked.